File: Client/supermarket.js
import { Red } from "./red.js"
import { Proizvod } from "./proizvod.js";
import { Skladiste } from "./skladiste.js"
import { SProizvod } from "./sproizvod.js";

export class Supermarket
{
    constructor(id, grad, ime)
    {
        this.redovi = [];
        this.skladista = [];
        this.id=id;
        this.grad=grad;
        this.ime=ime;
        this.container = null;
        this.rContainer = null; 
        this.sContainer = null; 
    }

}

File: Models/Supermarket.cs
using System.Collections.Generic;
using System.ComponentModel.DataAnnotations;
using System.ComponentModel.DataAnnotations.Schema;
using System.Text.Json.Serialization;

namespace Models
{
    [Table("Supermarket")]
    public class Supermarket
    {
        [Key]
        [Column("ID")]
        public int ID { get; set; }

        [Required]
        [Column("Ime")]
        [MaxLength(40)]
        public string Ime { get; set; }

        [Required]
        [Column("Grad")]
        [MaxLength(40)]
        public string Grad { get; set; }

        [JsonIgnore]
        public virtual List<Red> Redovi { get; set; }

        [JsonIgnore]
        public virtual List<Skladiste> Skladista { get; set; }
    }
}